sources/packages/api/src/services/student-appeal/student-appeal.service.ts

```diff
@@ -72,4 +79,53 @@ export class StudentAppealService extends RecordDataModelService<StudentAppeal>
 
     return !!existingAppeal;
   }
+
+  /**
+   * Get all pending and declined student Appeals
+   * for an application.
+   * Here we have added different when statement
+   * in CASE to fetch the status of the appeals.
+   * * WHEN: is checking if at least one Pending
+   * * appeals request is there for an appeal,
+   * * then the status is considered as Pending.
+   * * END: if any of them is not falling
+   * * under above case and applying  the andWhere
```

Comment:
Thanks for adding the comments explaining the details.
Please remove the extra white spaces in lines 92 and 94.